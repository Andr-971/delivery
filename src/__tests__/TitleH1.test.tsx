
import React from "react";
import { render, screen } from "@testing-library/react";
import TitleH1 from "../components/TitleH1/TitleH1";
import { TitleProps } from "../components/TitleH1/typeTitle";

describe("TitleH1 Component", () => {
    const testProps: TitleProps = {
        title: "Test Heading",
    };

    // Тест 1: Базовый рендеринг
    it('renders correctly with provided title', () => {
        render(<TitleH1 {...testProps} />);

        const headingElement = screen.getByTestId('title-h1');
        expect(headingElement).toBeInTheDocument();
        expect(headingElement).toHaveTextContent(testProps.title);
        expect(headingElement).toHaveClass('title__h1');
    });

    // Тест 2: Проверка HTML-тега
    it('renders correct h1 tag', () => {
        const { container } = render(<TitleH1 {...testProps} />);

        const h1Element = container.querySelector('h1.title__h1');
        expect(h1Element).not.toBeNull();
        expect(h1Element?.tagName).toBe('H1');
    });

    // Тест 3: Проверка содержимого
    it('displays dynamic content correctly', () => {
        const customTitle = 'Custom Header Text';
        render(<TitleH1 title={customTitle} />);

        expect(screen.getByText(customTitle)).toBeInTheDocument();
    });

    // Тест 4: Обработка пустого значения
    it('handles empty title gracefully', () => {
        render(<TitleH1 title="" />);

        const headingElement = screen.getByTestId('title-h1');
        expect(headingElement).toBeInTheDocument();
        expect(headingElement).toHaveTextContent('');
    });

    // Тест 5: Проверка структуры через snapshot
    it('matches snapshot', () => {
        const { asFragment } = render(<TitleH1 {...testProps} />);
        expect(asFragment()).toMatchSnapshot();
    });

    // Тест 6: Проверка отсутствия лишних элементов
    it('does not render unexpected elements', () => {
        const { container } = render(<TitleH1 {...testProps} />);

        expect(container.querySelector('div')).toBeNull();
        expect(container.querySelector('h2')).toBeNull();
        expect(container.querySelector('span')).toBeNull();
    });
});