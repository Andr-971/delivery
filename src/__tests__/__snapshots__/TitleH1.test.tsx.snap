// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleH1 Component matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="title__h1"
    data-testid="title-h1"
  >
    Test Heading
  </h1>
</DocumentFragment>
`;
